docs/core-concepts/app-types.md

```diff
@@ -0,0 +1,125 @@
+# Types of portability in .NET Core
+
+## Introduction
+There are several ways to think about the "types" of applications you can build. Usually, the types describe 
+a certain execution model or are based on what the application can "do"; examples of these are "console application", 
+"web application" etc. All of these types of applications (and more) can be created with .NET Core, since it is 
+a general purpose development platform.
+
+However, given its unique and cross-platform nature, .NET Core also has another angle through which to observe the type of the application 
+and that is application's *portability*. Portability here means, essentially, where can you run your application and what are the 
+prerequisites that you need to satisfy in order for your application to run on a given machine. This document 
+deals with this angle, portability, and outlines the two main types of portability that .NET Core enables. 
+
+There are two main types that we can observe: 
+
+1. Portable application
+    * As a sub-category of this we have the portable application with native dependencies
+2. Self-contained application
+
+## Portable application
+Portable applications are the default type in .NET Core. They require .NET Core to be installed on the targeted machine 
+in order for them to run. To you as a developer, this means that your application is portable between installations of 
+.NET Core. 
+
+This type of application will only carry its own code and dependencies that are outside of .NET Core libraries. 
+As an example, let's say you are making a console application that has the ability to invoke a certain REST API 
+and deserialize the returned JSON into a type and then display it. You have everything you need for this small 
+application expect for a good JSON parser; for this, you add a dependency to your `project.json` to include 
+[Json.NET](https://www.nuget.org/packages/Newtonsoft.Json/). Once you publish your application using `dotnet publish`, 
+you will see that only your application's code and JSON.net have been published in the output. 
+The .NET Core libraries remains outside of your application's dependency closure. 
+
+In order to create a portable application, all you need to do is to target the .NET Core libraries in your `project.json`
+and have your frameworks aligned as the below sample shows. 
+
+```json
+"dependencies": {
+    "Microsoft.NETCore.App": {  
+        "version": "1.0.0-*",
```

Comment:
Once we RTM, will this change to be a specific version #? I've heard that "floating versions" are not preferable in general.